### PR TITLE
Created initial interface for an Abstract Cloud.

### PR DESCRIPTION
**Pull request by ancord-m**

### CODE REVIEW

File: external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    //TODO what should be a parameter and a return value?
+    void downloadBinFile();
+
+    /**
+     * A createDirectory implementation method allows one to create a new folder or directory
+     * at a cloud storage.
+     * @return A method implementation should signal whether a directory was successfully created (true) or not (false).
+     */
+    //FIXME param?
+    boolean createDirectory();
```
Comment (vasap87) [inline external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: For example yandex disk api method to create directory has String parameter with path of creating folder. Response was link of creating folder.
My opinion - we pass String with path and get String with URL. This response URL we can use to generate VUSF snapshot.

File: external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * This method implementation is similar with createDirectory but removes a certain directory.
+     * @return A method implementation should return true if a directory was remove and false otherwise.
+     */
+    //FIXME param?
+    boolean removeDirectory();
```
Comment (vasap87) [inline external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: Param can be a String with path of removed directory.
Why we devide directory and file?

File: external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package com.stnetix.ariaddna.externalcloudapi.cloudinterface;
+
+/**
+ * This is an abstract interface containing general methods for manipulation with any external cloud storage.
+ * Every certain cloud storage realization must implement this interface.
+ */
+public interface iAbstractCloud {
```
Comment (vasap87) [inline external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: I think we need next methods:
String moveFile(String from, String to)
String copyFile(String from, String to)

File: external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 2 above)
Comment (slonikmak) [inline external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: for file systems is no difference between a file and a directory, and probably easier to make a single interface for working with them. And in the VUFS, I suggest using the same approach, checking if necessary the isDirectory(). I also want to suggest using Path as a unique identifier for the file and a means of accessing and manipulating it.

File: external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 3 above)
Comment (ancord-m) [inline external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: I've added both

File: external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java (hunk 1 above)
Comment (ancord-m) [inline external-cloud-api/src/com/stnetix/ariaddna/externalcloudapi/cloudinterface/iAbstractCloud.java]: I've changed. Now it should return String with a URL.

**Pull request closed without merging**